### DISCUSSION TITLE
Lesson 0: Kind of not clear what a node is

### DISCUSSION THREAD

**Discussion by chetanpandey1266** [smartcontractkit/full-blockchain-solidity-course-py#1462] [Q&A] [answered]
Node is a system which has a copy of whole blockchain. And there are many such nodes. I think I am right upto this point.
Now the first question is that who are these nodes? Like I am making a smart contract and posting on ethereum blockchain so will I be become a node?
Second question is that if we are moving from centralised system to decentralised system, then will our space requirement will also increase? If this is so then we can't use blockchain for systems with lot of data.

**Comment by cromewar** [comment, marked as answer]
Hello @chetanpandey1266 
Nice questions

>  who are these nodes? Like I am making a smart contract and posting on ethereum blockchain so will I be become a node?

No, you are just using a service provided by the nodes to store that smart contract, a node is a person or group of persons which uses their computers as databases for other people, such as you or me. Because of that, we as the users pay a transaction fee which is the payment the node receives for their services. Also, once a new block is created the node who made the register, earns an extra reward known as block reward.

> Second question is that if we are moving from centralized system to decentralized system, then will our space requirement will also increase? If this is so then we can't use blockchain for systems with lot of data.


Yes this is actually one problematic and controversial point about blockchain as a whole, as the data is decentralized there are a lot of redundancy, and the data is stored simultaneously on different computers around world, so is energetic expensive. 
Theoretically you can use the blockchain to store any amount of data, but practically is not viable, the higher the file size the expensive the cost would be, for you as user paying for the transaction and for the system itself to support that data, the convention on this case is to just store the most valuable/important an relevant information on the blockchain and left the other information centralized outside it, that's why there are a lot of services like Moralis trying to fulfill that needs.

At the end there is IPFS which is a decentralized way to store information (kind of) but is not 100% decentralized at the end.

**Reply by cromewar** [reply]
@PatrickAlphaC Am I missing something else here?

**Reply by chetanpandey1266** [reply]
Thanks @cromewar for this detailed answer.
